src/ui/widgets/XYPlot/xyPlot.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { CSSProperties } from "react";
+import { Widget } from "../widget";
+import {
+  InferWidgetProps,
+  ColorPropOpt,
+  StringPropOpt,
+  FontPropOpt,
+  TracesPropOpt,
+  AxesPropOpt,
+  BoolPropOpt,
+  FloatPropOpt
+} from "../propTypes";
+import { PVComponent, PVWidgetPropType } from "../widgetProps";
+import { registerWidget } from "../register";
+import Plot from "react-plotly.js";
+import {
+  calculateAxisLimits,
+  createAxes,
+  createTraces,
+  NewAxisSettings
+} from "./xyPlotOptions";
+import { Color } from "../../../types/color";
+
+export const XYPlotProps = {
+  height: FloatPropOpt,
+  width: FloatPropOpt,
+  plotBackgroundColor: ColorPropOpt,
+  title: StringPropOpt,
+  titleFont: FontPropOpt,
+  showLegend: BoolPropOpt,
+  showPlotBorder: BoolPropOpt,
+  showToolbar: BoolPropOpt,
+  traces: TracesPropOpt,
+  axes: AxesPropOpt
+};
+
+export type XYPlotComponentProps = InferWidgetProps<typeof XYPlotProps> &
+  PVComponent;
+
+export const XYPlotComponent = (props: XYPlotComponentProps): JSX.Element => {
+  const {
+    height = 250,
+    width = 400,
+    value,
+    plotBackgroundColor = Color.fromRgba(255, 255, 255),
+    title = "",
+    titleFont,
+    showLegend = true,
+    showPlotBorder,
+    // showToolbar, // TO DO - do we want a toolbar as well?
+    traces,
+    axes
+  } = props;
+  // TO DO - having all these checks is not ideal
+  if (
+    value?.value.arrayValue &&
```

Comment:
Will we only support waveform PVs? In CS-Studio the xyplot also plots 'live' data as the PV value changes? I guess this will need the concatenate data ability which I know is not working yet.

Reply:
We would want to support live data plots too, which I assume would be plotting the live data against time as it comes in? You're correct that it would use the concatenate data, and also reconfiguring to have time as an axis instead of index. I believe the plan is to add in support for time series plots (and also multiple PVs for a single widget), but I haven't got around to it quite yet....